switcher.js

```diff
@@ -35,7 +35,16 @@ class Switcher {
       throw new Error(`Incorrectly formatted credentials file found in ${this.awsDir}`);
     }
 
-    this.profileData = this._getCredentials(this.awsDir);
+    const configResult = this._checkConfigFile(this.awsDir);
+    if (configResult === 'good') {
+      this.processConfig = true;
+    } else {
+      this.processConfig = false;
+    }
+
+    this.profileData = this._getCredentials().then(profileData => {
```

Comment:
It would probably be better to combine these differently, since it's possible that there would be collisions in the keys between the credentials file and the config file. Maybe having this data as:

```js
{
  profileData: {},
  configData: {},
}
```

or maybe just having a separate property on `this` to hold the data from the config file. I think attaching a new `this` property might end up cleaner, and might allow better separation of the logic where necessary.